sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -187,6 +188,162 @@ case class Rint(child: Expression) extends UnaryMathExpression(math.rint, "ROUND
   override def funcName: String = "rint"
 }
 
+abstract class BitwiseShiftExpression
+  extends Expression with trees.BinaryNode[Expression] {
+  self: Product =>
+
+  def symbol: String = sys.error(s"BitwiseShiftExpression must override symbol")
+
+  override def nullable: Boolean = true
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (!Seq(ByteType, ShortType, IntegerType, LongType, NullType).contains(left.dataType)
+      || !(right.dataType.isInstanceOf[IntegerType] || right.dataType == NullType)
+    ) {
+      TypeCheckResult.TypeCheckFailure(
+        s"Incorrect data types for $symbol(${left.dataType}, ${right.dataType}})")
+    } else {
+      TypeCheckResult.TypeCheckSuccess
+    }
+  }
+
+  override def dataType: DataType = left.dataType match {
+    case ByteType | ShortType | IntegerType => IntegerType
+    case LongType => LongType
+    case NullType => NullType
```

Comment:
In Hive, seems it convert the `NullType` to `ByteType`. See https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/NumericOpMethodResolver.java#L90